src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/TagHelperResolver.cs

```diff
@@ -47,12 +57,21 @@ protected virtual async Task<TagHelperResolutionResult> GetTagHelpersAsync(Proje
                 context.SetCompilation(compilation);
             }
 
+            var timingDictionary = new Dictionary<string, long>();
             for (var i = 0; i < providers.Length; i++)
             {
                 var provider = providers[i];
+                var stopWatch = Stopwatch.StartNew();
+
                 provider.Execute(context);
+
+                stopWatch.Stop();
+                var propertyName = $"razor.{provider.Name}.elapsedtimems";
+                Debug.Assert(!timingDictionary.ContainsKey(propertyName));
+                timingDictionary[propertyName] = stopWatch.ElapsedMilliseconds;
             }
 
+            _telemetryReporter.ReportEvent("taghelperresolver/gettaghelpers", VisualStudio.Telemetry.TelemetrySeverity.Normal, timingDictionary.ToImmutableDictionary());
```

Comment:
I don't know anything about tag helper providers, but can we get a count of how many tag helpers each one found? might be interesting